pkg/operator/staticpod/controller/monitoring/monitoring_resource_controller.go

```diff
@@ -0,0 +1,204 @@
+	"time"
+
+	"github.com/golang/glog"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	corelisterv1 "k8s.io/client-go/listers/core/v1"
+	rbaclisterv1 "k8s.io/client-go/listers/rbac/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/assets"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/staticpod/controller/common"
+	"github.com/openshift/library-go/pkg/operator/staticpod/controller/monitoring/bindata"
+)
+
+const (
+	defaultMonitoringNamespace                        = "openshift-monitoring"
+	operatorStatusMonitoringResourceControllerFailing = "MonitoringResourceControllerFailing"
+	controllerWorkQueueKey                            = "key"
+	manifestDir                                       = "pkg/operator/staticpod/controller/monitoring"
+)
+
+type MonitoringResourceController struct {
+	targetNamespace    string
+	serviceMonitorName string
+
+	saListerSynced cache.InformerSynced
+	saLister       corelisterv1.ServiceAccountLister
+
+	clusterRoleBindingLister       rbaclisterv1.ClusterRoleBindingLister
+	clusterRoleBindingListerSynced cache.InformerSynced
+
+	// queue only ever has one item, but it has nice error handling backoff/retry semantics
+	queue workqueue.RateLimitingInterface
+
+	kubeClient           kubernetes.Interface
+	dynamicClient        dynamic.Interface
+	operatorConfigClient common.OperatorClient
+	eventRecorder        events.Recorder
+}
+
+// NewMonitoringResourceController creates a new backing resource controller.
+func NewMonitoringResourceController(
+	targetNamespace string,
+	serviceMonitorName string,
+	operatorConfigClient common.OperatorClient,
+	kubeInformersForTargetNamespace informers.SharedInformerFactory,
+	kubeClient kubernetes.Interface,
+	dynamicClient dynamic.Interface,
+	eventRecorder events.Recorder,
+) *MonitoringResourceController {
+	c := &MonitoringResourceController{
+		targetNamespace:      targetNamespace,
+		operatorConfigClient: operatorConfigClient,
+		eventRecorder:        eventRecorder,
+		serviceMonitorName:   serviceMonitorName,
+
+		clusterRoleBindingListerSynced: kubeInformersForTargetNamespace.Core().V1().ServiceAccounts().Informer().HasSynced,
+		clusterRoleBindingLister:       kubeInformersForTargetNamespace.Rbac().V1().ClusterRoleBindings().Lister(),
+
+		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "MonitoringResourceController"),
+		kubeClient:    kubeClient,
+		dynamicClient: dynamicClient,
+	}
+
+	operatorConfigClient.Informer().AddEventHandler(c.eventHandler())
+
+	// TODO: We need a dynamic informer here to observe changes to ServiceMonitor resource.
```

Comment:
@sttts @deads2k right now, when something stomp the service monitor or delete it, we won't notice right away, but in next reconcile cycle for this controller (or when resources we monitor change)... 

To fix this, we will need dynamic informers which will be available in 1.13. Alternatively, as @sttts suggested, we can copy dynamicinformer package into library-go and then replace it. However, I don't think it is worth the effort for now and I can create follow up issue to add the informer when we get 1.13 ?

Reply:
> @sttts @deads2k right now, when something stomp the service monitor or delete it, we won't notice right away, but in next reconcile cycle for this controller (or when resources we monitor change)...

Just set a poll every few minutes for this controller.